packages/grid/x-grid/src/tests/editRows.XGrid.test.tsx

```diff
@@ -254,4 +255,33 @@ describe('<XGrid /> - Edit Rows', () => {
     fireEvent.click(input0);
     expect(document.activeElement).to.have.property('value', 'Nike');
   });
+
+  it('should apply the valueParser before saving the value', () => {
+    const valueParser = stub().withArgs('62').returns(1962);
+    render(
+      <TestCase
+        columns={[
+          { field: 'brand', editable: true },
+          { field: 'year', editable: true, valueParser },
+        ]}
+      />,
```

Comment:
We don't need the apiRef, we might want to avoid one attraction for clarity:

```suggestion
      <XGrid
        {...baselineProps}
        columns={[
          { field: 'brand', editable: true },
          { field: 'year', editable: true, valueParser },
        ]}
      />,
```